### PR TITLE
fix: prevent parser generation error on tree-sitter 0.24.3

### PR DESCRIPTION
**Pull request by ribru17**
Parser generation fails on tree-sitter 0.24.3 due to unescaped curly brackets in a regex pattern. This commit escapes them and runs `tree-sitter g` on the latest version

### CODE REVIEW

**Comment by ribru17** [comment]
cc @sdankel :)

**Comment by clason** [comment]
You should probably update your tests, too -- Node 14 has been removed quite a while ago... ;)

**Comment by sdankel** [comment]
@ribru17 Could you expand on the issue, and add test cases for it? Why are there so many changes in parser.c? Also, please run the tests locally and make sure there's passing.

**Comment by clason** [comment]
@sdankel 

1. The issue is that tree-sitter upstream changed the interpretation of regex (aligning with Rust regex rather than the weird mixture of JS and Rust it had before). We fixed a lot of parsers, but this one was only recently added to nvim-treesitter so we missed it.
2. The changes in the generated files are due to changes in tree-sitter CLI (the parser here was very old); most of them are typical "you change one entry in a table, you change the whole table" tree-sitter churn, but there was also changes (performance optimizations and fixes) in how some symbols are handled.
3. Test failure is due to outdated workflows (using Node 14, which has been removed from runners); I can confirm that `tree-sitter test` passes on that branch.

**Comment by clason** [comment]
Not sure about the sway repo failures; since the last test was run a year ago, it's hard to tell if it's due to changes in this PR, breaking tree-sitter changes, or changes in the corpus (you're not pinning a commit, so you're at the mercy of the sway repo not to add breaking changes).

**Comment by clason** [comment]
@sdankel you might be interested in https://github.com/tree-sitter/workflows

File: tree-sitter.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    }
+  },
+  "bindings": {
+    "c": true,
+    "go": true,
+    "node": true,
+    "python": true,
+    "rust": true,
+    "swift": true
+  }
```
Comment (sdankel) [inline tree-sitter.json:35]: Shouldn't the rust binding be the only one set to true?

Comment (clason) [reply tree-sitter.json:35]: And `node`, I'd say.

Comment (sdankel) [reply tree-sitter.json:35]: It looks like `tree-sitter generate` should generate these bindings. Perhaps we should leave them set to true and also check in the generated bindings.

**Comment by sdankel** [comment]
@clason Thanks for the detailed explanation and suggestions. For now I've pinned the sway language to an older version to unblock CI.

@ribru17 If you could please rebase your branch and apply the suggested changes, and rerun `tree-sitter generate` with the latest version of the tree-sitter CLI, I think this should be good to go. Thanks for bringing this to my attention and for the contribution.

File: tree-sitter.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    "license": "Apache-2.0",
+    "description": "Sway grammar for tree-sitter",
+    "authors": [
+      {
+        "name": "Fuel Labs",
+        "email": "contact@fuel.sh"
+      }
+    ],
+    "links": {
+      "repository": "git+https://github.com/FuelLabs/tree-sitter-sway.git"
```
Comment (sdankel) [inline tree-sitter.json:25]:
```suggestion
      "repository": "https://github.com/FuelLabs/tree-sitter-sway"
```

File: tree-sitter.json (hunk 1 above)
Comment (clason) [reply tree-sitter.json:35]: Yes, it does; but people often complain (understandably) if a bugfix PR comes with a metric butt-ton of weird files and bindings ;) So I imagine this PR intentionally omitted most "optional" files (even though one or two related changes slipped through).

**Comment by ribru17** [comment]
I have applied the above changes, and checked in the generated bindings per your suggestion (in a separate commit for clarity, feel free to squash). Thanks for reviewing!

**Review by sdankel — APPROVED** [review]

**Pull request merged**